pkg/api/types.go

```diff
@@ -2154,3 +2154,38 @@ type RangeAllocation struct {
 	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
 	Data []byte `json:"data"`
 }
+
+// LockSpec is the specification of a resource lock.
+type LockSpec struct {
+	// Who currently has the lock.
+	HeldBy string `json:"heldBy"`
+	// How long the lock is valid.
+	LeaseSeconds uint64 `json:"leaseSeconds"`
+}
+
+type LockStatus struct {
+	// When the lock was first acquired.
+	AcquiredTime util.Time `json:"acquiredTime,omitempty"`
+	// When the lock was last renewed.
+	RenewTime util.Time `json:"renewTime,omitempty"`
+}
+
+// Lock represents the configuration of a resource lock.
+type Lock struct {
```

Comment:
Which part? The namespace controller specifically, or the more general client library issue?

As for putting this in v1: Help with the experimental API/client instead.